compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1889,40 +1884,41 @@ private void recursivelySetFinalFlag(BLangVariable variable) {
 
     // Statements
 
-    public void visit(BLangBlockStmt blockNode) {
-        env = SymbolEnv.createBlockEnv(blockNode, env);
+    public void visit(BLangBlockStmt blockNode, AnalyzerData data) {
+        SymbolEnv symbolEnv = SymbolEnv.createBlockEnv(blockNode, data.env);
+        data.env = symbolEnv;
         int stmtCount = -1;
         for (BLangStatement stmt : blockNode.stmts) {
             stmtCount++;
             boolean analyzedStmt = analyzeBlockStmtFollowingIfWithoutElse(stmt,
-                    stmtCount > 0 ? blockNode.stmts.get(stmtCount - 1) : null, env, false);
+                    stmtCount > 0 ? blockNode.stmts.get(stmtCount - 1) : null, data.env, false, data);
```

Comment:
Since we already pass `data` can we reduce number of arguments here

Reply:
In L503(calling `analyzeBlockStmtFollowingIfWithoutElse`) we use generated `env` instead of current `env`